topology/probes/sriov/sriov.go

```diff
@@ -0,0 +1,172 @@
+// +build linux
+
+/*
+ * Copyright (C) 2018 Samsung, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package sriov
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/vishvananda/netlink"
+
+	"github.com/skydive-project/skydive/common/observers/fs"
+	"github.com/skydive-project/skydive/filters"
+	"github.com/skydive-project/skydive/logging"
+	"github.com/skydive-project/skydive/topology"
+	"github.com/skydive-project/skydive/topology/graph"
+)
+
+// Probe is a Sriov probe designed to handle all vfs on server
+type Probe struct {
+	Graph   *graph.Graph
+	Root    *graph.Node
+	Cancel  context.CancelFunc // cancel function
+	watcher *fs.Watcher
+}
+
+// MANAGER is a const to be used as a string to describe who handles sriov nodes
+const MANAGER = "vfio"
+
+// GenerateNameForNodeFromPhysInterfaceNameAndVFID - generates a name for vf based on physical device name and id of of this vf
```

Comment:
nit: this seems to be used only once (and the name is a bit long :-) ) so I suggest we just inline it in the function where it is used

Reply:
Later on it will be used in libvirt probe to connect sriov vf with libvirt vm. I'll make pull request later.

Reply:
ok